README.md

```diff
@@ -42,6 +42,117 @@ Crystalball.foresee do |predictor|
 end
 ```
 
+## Map Generator
+
+There are different map generator strategies that can (and should) be used together for better predictions. Each one has its own benefits and drawbacks, so they should be configured to best fit your needs.
+
+### CoverageStrategy
+
+Uses coverage information to detect which files are covered by the given spec (i.e. the files that, if changed, may potentially break the spec);
+To customize the way the execution detection works, pass an object that responds to #detect and returns the paths to the strategy initialization:
+
+```ruby
+# ...
+config.register Crystalball::MapGenerator::CoverageStrategy.new(MyDetector)
+```
+
+By default, the execution detector is a `Crystalball::MapGenerator::CoverageStrategy::ExecutionDetector`, which filters out the paths outside the root and converts absolute paths to relative.
+
+### AllocatedObjectsStrategy
+
+Looks for the files in which the objects created during the spec execution are defined. It is considerably slower than `CoverageStrategy`.
+To use this strategy, use the convenient method `.build` which takes two optional keyword arguments: `only`, used to define the classes or modules to have their descendants tracked (defaults to `[]`); and `root`, which is the path where the detection will take place (defaults to `Dir.pwd`).
+Here's an example that tracks allocation of `ActiveRecord::Base` objects:
+
+```ruby
+# ...
+config.register Crystalball::MapGenerator::AllocatedObjectsStrategy.build(only: ['ActiveRecord::Base'])
+```
+
+That method is fine for most uses, but if you need to further customize the behavior of the strategy, you can directly instantiate the class.
+
+```ruby
+# ...
+config.register Crystalball::MapGenerator::AllocatedObjectsStrategy
+  .new(execution_detector: MyCustomDetector, object_tracker: MyCustomTracker)
+```
+
+The initialization takes two keyword arguments: `execution_detector` and `object_tracker`.
+`execution_detector` must be an object that responds to `#detect` receiving a list of objects and returning the paths affected by said objects. `object_tracker` is something that responds to `#created_during` which yields to the caller and returns the array of objects allocated during the execution of the block.
+
+### DescribedClassStrategy
+
+This strategy will take each example that has a `described_class` (i.e. examples inside `describe` blocks of classes and not strings) and add the paths where the described class and its ancestors are defined to the case map of the example;
+
+To use it, add to your `Crystalball::MapGenerator.start!` block:
+
+```ruby
+# ...
+config.register Crystalball::MapGenerator::DescribedClassStrategy.new
+```
+
+As with `CoverageStrategy`, you can pass a custom execution detector (an object that responds to `#detect` and returns the paths) to the initialization:
```

Comment:
Execution detector for DescribedClassStrategy is more like for AllocatedObjectsStrategy